Lib/test/test_asyncio/test_unix_events.py

```diff
@@ -1911,8 +1913,14 @@ def test_fork_signal_handling(self):
         parent_handled = manager.Event()
 
         def child_main():
-            signal.signal(signal.SIGTERM, lambda *args: child_handled.set())
+            def on_sigterm(*args):
+                child_handled.set()
+                sys.exit()
+
+            signal.signal(signal.SIGTERM, on_sigterm)
             child_started.set()
+            while True:
+                time.sleep(1)
```

Comment:
It took me a while to understand that the child doesn't run the loop forever, but is interrupted by SIGTERM and so exits immediately.